Lib/asyncio/runners.py

```diff
@@ -114,15 +132,17 @@ def run(self, coro, *, context=None):
 
         self._interrupt_count = 0
         try:
-            if self._set_event_loop:
-                events.set_event_loop(self._loop)
-            return self._loop.run_until_complete(task)
+            if self._set_policy_loop:
+                events.set_event_loop(loop)
+            return loop.run_until_complete(task)
         except exceptions.CancelledError:
             if self._interrupt_count > 0 and task.uncancel() == 0:
                 raise KeyboardInterrupt()
             else:
                 raise  # CancelledError
         finally:
+            if self._set_policy_loop:
+                events.set_event_loop(None)
```

Comment:
`asyncio.Runner` should be used a context manager and its close method removes the set event loop so not required here.

Reply:
```python
with Runner() as runner1, Runner() as runner2:
    runner1.run(fn())  # calls set_event_loop
    # I'd expect the event loop to be unset here
```

Reply:
>  I'd expect the event loop to be unset here

That happens only if you supply your `loop_factory` otherwise for existing code, its better to keep it set for existing code not using `asyncio.Runner`.

Reply:
I would expect the event loop to be set until the context manager exit which is what we have currently.